File: Design_Pattern_and_Principles/ObserverPatternExample/ObserverTest.java
public class ObserverTest {
    public static void main(String[] args) {
        StockMarket market = new StockMarket();

        MobileApp app = new MobileApp();
        WebApp web = new WebApp();

        market.registerObserver(app);
        market.registerObserver(web);

        System.out.println("Broadcasting market ticker updates...");
        market.updatePrice("AAPL", 185.50);
    }
}

File: Week_1/Design_Pattern_and_Principles/ObserverPatternExample/ObserverTest.cs
using System;

public class ObserverTest {
    public static void Main(string[] args) {
        StockMarket market = new StockMarket();

        MobileApp app = new MobileApp();
        WebApp web = new WebApp();

        market.RegisterObserver(app);
        market.RegisterObserver(web);

        Console.WriteLine("Broadcasting market ticker updates...");
        market.UpdatePrice("AAPL", 185.50);
    }
}